scripts/upstreamMerge.sh

```diff
@@ -1,41 +1,44 @@
 #!/usr/bin/env bash
 
-(
-set -e
-PS1="$"
-basedir="$(cd "$1" && pwd -P)"
-workdir="$basedir/work"
-gitcmd="git -c commit.gpgsign=false"
+set -euo pipefail
 
-updated="0"
-function getRef {
-    git ls-tree $1 $2  | cut -d' ' -f3 | cut -f1
+function git {
+  command git -c commit.gpgsign=false "${@}"
 }
+
+readonly basedir="$(cd "$(dirname "$0")/.." && pwd -P)"
+readonly workdir="${basedir}/work"
+
 function update {
-    cd "$workdir/$1"
-    $gitcmd fetch && $gitcmd clean -fd && $gitcmd reset --hard origin/master
-    refRemote=$(git rev-parse HEAD)
-    cd ../
-    $gitcmd add --force $1
-    refHEAD=$(getRef HEAD "$workdir/$1")
-    echo "$1 $refHEAD - $refRemote"
-    if [ "$refHEAD" != "$refRemote" ]; then
-        export updated="1"
-    fi
+  local -r submodule="${1?Submodule is missing}"
+  local -r submodule_dir="${workdir}/${submodule}"
+  cd "${submodule_dir}"
+  git fetch && git clean -fd && git reset --hard origin/master || return 1
+  cd "${basedir}"
+  git add --force "${submodule_dir}"
 }
 
-update Bukkit
-update CraftBukkit
-update Spigot
-
-if [[ "$2" = "all" || "$2" = "a" ]] ; then
-	update BuildData
-fi
-if [ "$updated" == "1" ]; then
+function main {
+  local -r update_all="${1:-}"
+  local -r skip_rebuild="${2:-}"
+  cd "${basedir}"
+  git submodule update --init
+  update Bukkit
+  update CraftBukkit
+  update Spigot
+  if [[ "${update_all}" == @(all|a|yes|y) ]]; then
```

Comment:
```
 15:21:29  shane@Shanes-iMac  ...work/paper/Paper   update_scripts ✔ 
$ ./scripts/upstreamMerge.sh
./scripts/upstreamMerge.sh: line 29: syntax error in conditional expression: unexpected token `('
```

Reply:
Fixed this. Tested with bash 3.2.57 and found a bug with `command` builtin, so I also removed that part of the code. Seems to work fine now.